src/intra_edge.rs

```diff
@@ -144,38 +157,27 @@ unsafe fn init_mode_node(
     top_has_right: bool,
     left_has_bottom: bool,
 ) {
-    init_edges(
-        &mut nwc.node,
+    // this is an invariant: this function is only called with BL_128X128 and BL_64X64.
+    // then this function recurses, decreasing the block level at each step, until bottoming out
+    // at BL_16x16.
+    assert!(bl != BL_8X8);
```

Comment:
I think this logical step is required to make the logic safe here. It is a change to the logic (versus the C code) but hopefully it's close enough.